apps/api/src/donations/donations.service.ts

```diff
@@ -546,29 +547,33 @@ export class DonationsService {
     return donation
   }
 
-  async createManyBankPayments(donationsDto: CreateManyBankPaymentsDto[]) {
-    for (const donation of donationsDto) {
-      await this.prisma.$transaction(async (tx) => {
-        //to avoid incrementing vault amount twice we first check if there is such donation
-        const existingDonation = await tx.donation.findUnique({
-          where: { extPaymentIntentId: donation.extPaymentIntentId },
+  async createUpdateBankPayment(donationsDto: CreateManyBankPaymentsDto): Promise<ImportStatus> {
```

Comment:
I see the function is refactored to work on a single donation - could you rename the input parameter from plural to singular too

Reply:
I agree that this should be renamed. We have a similar object **CreateBankPaymentDto** https://github.com/podkrepi-bg/api/blob/master/apps/api/src/donations/dto/create-bank-payment.dto.ts . 
From my investigation , I saw that we expose it through the following endpoint `/create-bank-payment`, but I couldn't find where it is used. My assumption is that the object is outdated. 
If this is the case I can refactor **CreateBankPaymentDto** with the proper fields and reuse it instead **CreateManyBankPaymentsDto**. Would you please confirm if this object and endpoint is not in use?

Reply:
well spotted! 
The CreateBankPaymentDto and  CreateManyBankPaymentsDto should be the same as they are used for the same thing. Indeed the CreateBankPaymentDto is outdated - it is used for manual creation of bankImports in the same Admin UI where the file import happens. Please take the new fields from CreateManyBankPaymentsDto and let's unify them into one.
Btw. feel free to do this in separate PR because it will probably require more changes in frontend too

Reply:
Alright! I would make a new PR after this one is  being merged, where I will unify the structures.